### DISCUSSION TITLE
Permission to use GraphQL API with GITHUB_TOKEN on GitHub Actions

### DISCUSSION THREAD

**Discussion by peaceiris** [community/community#5082] [Actions]
My proposal about a REST API to toggle pull-request draft status ([pulls.update: Change pull-request draft status · Issue #69 · octokit/rest.js](https://github.com/octokit/rest.js/issues/69)) has been rejected since the GitHub API v4 (GraphQL API) has the ability to do that.

Unfortunately, an auto-generated access token called `GITHUB_TOKEN` seems not to have full control permission of the GraphQL API. We got the error `Resource not accessible by integration` with the GitHub API v4 and `GITHUB_TOKEN`.

This pull-request uses a personal access token with a `public_repo` scope (for a public repository) as a workaround.

Related:

- https://github.com/peaceiris/actions-label-commenter/pull/485
- https://github.com/peaceiris/actions-label-commenter/pull/487
- https://github.com/github/feedback/discussions/5081

**Comment by Andre601** [comment]
Hopping in to say that this is quite annoying.

Members with manage issue/PR rights can to my knowledge change a PRs status from/to draft and the standard Github token should grant similar privileges since closing and/or locking are available through it too.

While using your own access token with sufficient rights is a working workaround does it come with the severe downside of using your personal account (or whichever was used to generate the token) for the action, which is something I and probably many others do not want for reasons such as security.
There is really no reason for the GitHub token to not also have rights to change a PRs draft status and I honestly see this as a possible bug even, so a fix for this would be really nice to have.
